internal/controller/utils.go

```diff
@@ -285,3 +287,16 @@ func SetDefaults_Deployment(obj *appsv1.Deployment) {
 		*obj.Spec.ProgressDeadlineSeconds = 600
 	}
 }
+
+func getProxyEnvVars() []corev1.EnvVar {
+	envVars := []corev1.EnvVar{}
+	for _, envvar := range []string{"HTTPS_PROXY", "https_proxy", "HTTP_PROXY", "http_proxy", "NO_PROXY", "no_proxy"} {
+		if value := os.Getenv(envvar); value != "" {
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  strings.ToLower(envvar),
```

Comment:
@syedriko Do we want the ToLower here , the example in the docs is in upper case
https://docs.openshift.com/container-platform/4.15/operators/admin/olm-configuring-proxy-support.html

Reply:
We do :) https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ is the definitive source. I stole that ^^^ code is from the cluster logging operator.